x-pack/filebeat/input/httpjson/response.go

```diff
@@ -74,63 +74,68 @@ func newResponseProcessor(config *responseConfig, pagination *pagination, log *l
 	return rp
 }
 
-func (rp *responseProcessor) startProcessing(stdCtx context.Context, trCtx *transformContext, resp *http.Response) (<-chan maybeMsg, error) {
+func (rp *responseProcessor) startProcessing(stdCtx context.Context, trCtx *transformContext, resp []*http.Response) (<-chan maybeMsg, error) {
 	ch := make(chan maybeMsg)
 
 	go func() {
 		defer close(ch)
 
-		iter := rp.pagination.newPageIterator(stdCtx, trCtx, resp)
-		for {
-			page, hasNext, err := iter.next()
-			if err != nil {
-				ch <- maybeMsg{err: err}
-				return
-			}
+		for i, httpResp := range resp {
+			iter := rp.pagination.newPageIterator(stdCtx, trCtx, httpResp)
+			for {
+				page, hasNext, err := iter.next()
+				if err != nil {
+					ch <- maybeMsg{err: err}
+					return
+				}
 
-			if !hasNext {
-				return
-			}
+				if !hasNext {
+					if i+1 != len(resp) {
+						break
+					}
+					return
+				}
 
-			respTrs := page.asTransformables(rp.log)
+				respTrs := page.asTransformables(rp.log)
 
-			if len(respTrs) == 0 {
-				return
-			}
+				if len(respTrs) == 0 {
+					return
+				}
 
-			trCtx.updateLastResponse(*page)
+				trCtx.updateLastResponse(*page)
 
-			rp.log.Debugf("last received page: %#v", trCtx.lastResponse)
+				rp.log.Debugf("last received page: %#v", trCtx.lastResponse)
```

Comment:
[Question]

Should we add some information about the request in the chain?

Reply:
I am not sure but what do you think, should we add like chain URL(generated using ids)?

Reply:
Maybe, or the index from the for loop. The main idea is that when looking at the logs, we know from which request in the chain this log entry was about.